pkg/client/client.go

```diff
@@ -39,11 +55,25 @@ func CreateXrayClient(region string, datasourceInfo *awsds.AWSDatasourceSettings
 }
 
 // CreateEc2Client creates client for EC2 api. We need this for some introspection queries like getting regions.
-func CreateEc2Client(region string, datasourceInfo *awsds.AWSDatasourceSettings) (*ec2.EC2, error) {
-	sess, err := sessions.GetSession(region, *datasourceInfo)
+func CreateEc2Client(awsSettings *awsds.AWSDatasourceSettings, backendSettings *backend.DataSourceInstanceSettings) (*ec2.EC2, error) {
+	if awsSettings == nil {
+		return nil, fmt.Errorf("x-ray datasource settings required")
+	}
+	httpClientProvider := httpclient.NewProvider()
+	httpClientOptions, err := backendSettings.HTTPClientOptions()
+	if err != nil {
+		return nil, err
+	}
+	httpClient, err := httpClientProvider.New(httpClientOptions)
 	if err != nil {
+		backend.Logger.Error("failed to create HTTP client", "error", err.Error())
 		return nil, err
 	}
+	sess, err := sessions.GetSession(awsds.SessionConfig{
+		Settings:      *awsSettings,
+		HTTPClient:    httpClient,
+		UserAgentName: aws.String("X-ray"),
```

Comment:
same about the UserAgent being set later (you should be able to remove the function `userAgentString`